src/Orleans.Runtime/Transactions/TransactionAgent.cs

```diff
@@ -274,11 +302,87 @@ public bool IsAborted(long transactionId)
 
         }
 
+        private async Task CommitTransactions(List<TransactionInfo> committingTransactions,
+            HashSet<long> outstandingCommits)
+        {
+            var stopWatch = Stopwatch.StartNew();
+            try
+            {
+                metrics.BatchCommitTransactionsRequestsCounter++;
+                var commitResponse = await this.tmService.CommitTransactions(committingTransactions, outstandingCommits);
```

Comment:
To get a handle on the batching behavior, which is the point of all of this complexity, it would be good to track the average committingTransactions size per request.
Same for the start transaction operation.